src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the Duke chatbot.
+ *
+ */
+public class Duke {
+
+    protected static ArrayList<Task> arrayList = null;
+
+    private Ui ui;
+    private Parser parser;
+
+    /** Constructor for Duke */
+    private Duke() {
+        this.ui = new Ui();
+        this.parser = new Parser();
+    }
+
+    /**
+     * Runs the scanner and Duke chatbot.
+     *
+     */
+    private void run() throws DukeException, IOException {
+
+        this.ui.displayLogo();
+        Scanner sc = new Scanner(System.in);
+
+        String[] strArray = sc.nextLine().split(" ");
+
+        while (!strArray[0].equals("bye")) {
+            this.parser.parseCommand(strArray);
+
+            strArray = sc.nextLine().split(" ");
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
```

Comment:
Perhaps we could abstract the `println()` statements of all classes to `Ui.java`, since `Ui.java` should be the one handling the user interface side of things?